### PR TITLE
Parse tome paramdefs on startup

### LINKED ISSUE #295: Creating new tasks doesn't prompt for parameters
**Describe the bug**

**To Reproduce**
image

**Expected behavior**
Should prompt for parameters based on the definitions.
_metadata.yml_
```yaml
name: Process list
description: Get a list of running processes
parameter_definitions:
  - name: cmd_substring
    type: string
```

**Screenshots**

**Desktop (please complete the following information):**

**Smartphone (please complete the following information):**

**Additional context**

### PR DESCRIPTION
**Pull request by hulto**
#### What type of PR is this?:
/kind bug

#### What this PR does / why we need it:
- Switches the tome name to use the metadata file instead of file name.
- Calls `SetParamDefs` during tome parsing
- Updated the example metadata.yml file with example paramdefs.
- Increases the max length of the param defs field form 256 bytes to 4GB

#### Which issue(s) this PR fixes:
Fixes #295

### CODE REVIEW

**Review by alexcote1 — COMMENTED** [review]
Looks good, glad that static hello world is no longer there

File: tavern/tomes/example/linux/test-implant
Comment (Jorgey7) [inline tavern/tomes/example/linux/test-implant:1]: What did meowware ever do to you smh

Comment (hulto) [reply tavern/tomes/example/linux/test-implant:1]: Broke my heart 💔

**Comment by KCarretto** [comment]
We can make the paramdefs be more readable in the yaml, e.g.

'''
paramdefs:
- name: msg
  label: Message
'''

**Comment by hulto** [comment]
> We can make the paramdefs be more readable in the yaml, e.g.
> 
> '''
> paramdefs:
> - name: msg
>   label: Message
> '''

I can play with it more but when I tried that ^ it doesn't parse right since I think everything is expecting a JSON string.
We could do some gross reparsing to turn the yaml into a JSON string.

**Comment by hulto** [comment]
Actually I think we should prioritize consistency between the standard tomes and uploaded tomes.
Since uploaded tomes require paramDefs be a JSON string we should keep the standard ones with the JSON string.

If we want to re-do paramDefs to not be a JSON string we can do that later but for now I think we should stick with the agreed on paramDefs format.

**Comment by hulto** [comment]
Gonna pass to @cmp5987 
Looks like the front end isn't passing paramdefs rn.
image

**Comment by hulto** [comment]
image
Working now had to pass the params the right way.

**Comment by hulto** [comment]
Issue with golang.
image
```
$ cat /tmp/port_scan/metadata.yml 
name: Port scan
description: Scan a list of ports in a list of CIDRs.
paramdefs: "[{\"name\":\"cidrs\",\"label\":\"CIDRs\",\"type\":\"string\",\"placeholder\":\"['192.168.0.0./24','10.102.9.11/32']\"},{\"name\":\"ports\",\"label\":\"Ports\",\"type\":\"list\",\"placeholder\":\"[22,80,443]\"},{\"name\":\"protocol\",\"label\":\"Protocol\",\"type\":\"string\",\"placeholder\":\"tcp\"},{\"name\":\"timeout\",\"label\":\"Timeout\",\"type\":\"int\",\"placeholder\":\"2\"]"%                                                                                                                                                    ```

**Comment by hulto** [comment]
- add a test to parse go that validates passing no "paramdefs" is valid.

**Comment by hulto** [comment]
@KCarretto 
Missing one thing before this PR is ready.
The MySQL field still doesn't seem to be accepting the larger input despite changing it in the schema/tome file.

**Comment by hulto** [comment]
Blocking on go generate getting fixed.

**Review by KCarretto — APPROVED** [review]

**Pull request merged**